deepfence_server/reporters/graph/topology_reporter.go

```diff
@@ -1182,5 +1184,92 @@ func NewNeo4jCollector(ctx context.Context) (TopologyReporter, error) {
 	return nc, nil
 }
 
+func GetTopologyDelta(ctx context.Context,
+	deltaReq model.TopologyDeltaReq) (model.TopologyDeltaResponse, error) {
+
+	deltaResp := model.TopologyDeltaResponse{}
+	deltaResp.Additions = make([]model.NodeIdentifier, 0)
+	deltaResp.Deletions = make([]model.NodeIdentifier, 0)
+
+	driver, err := directory.Neo4jClient(ctx)
+	if err != nil {
+		return deltaResp, err
+	}
+
+	session, err := driver.Session(neo4j.AccessModeRead)
+	if err != nil {
+		return deltaResp, err
+	}
+
+	tx, err := session.BeginTransaction(neo4j.WithTxTimeout(30 * time.Second))
+	if err != nil {
+		return deltaResp, err
+	}
+	defer tx.Close()
+
+	processRecords := func(isAdd bool, query string) error {
+		r, err := tx.Run(query, map[string]interface{}{})
+		if err != nil {
+			return err
+		}
+
+		records, err := r.Collect()
+		if err != nil {
+			return err
+		}
+
+		for _, record := range records {
+			nodeid := record.Values[0].(string)
+			nodeType := record.Values[1].(string)
+			if isAdd == true {
+				deltaResp.Additions = append(deltaResp.Additions,
+					model.NodeIdentifier{nodeid, nodeType})
+			} else {
+				deltaResp.Deletions = append(deltaResp.Additions,
+					model.NodeIdentifier{nodeid, nodeType})
+			}
+		}
+
+		return nil
+	}
+
+	nodeTypeQueryStr := "("
+	for idx, entity := range deltaReq.EntityTypes {
+		nodeTypeQueryStr = nodeTypeQueryStr + "n:" + entity
+		if idx < (len(deltaReq.EntityTypes) - 1) {
+			nodeTypeQueryStr = nodeTypeQueryStr + " OR "
+		} else {
+			nodeTypeQueryStr = nodeTypeQueryStr + ")"
+		}
+	}
+
+	threshold := deltaReq.Timestamp + (30 * 1000)
```

Comment:
Since we pass the timestamp, do we need a threshold?
We could leave that choice over the user, they can already choose their own threshold by modifying the timestamp they send.

If any resource consumption concerns, we can simply timeout the neo4j request directly in the `Run()` call

Reply:
Done.